src/scores/continuous/correlation/correlation_impl.py

```diff
@@ -58,3 +58,65 @@ def pearsonr(
     )
 
     return xr.corr(fcst, obs, reduce_dims)
+
+
+def spearmanr(
+    fcst: xr.DataArray,
+    obs: xr.DataArray,
+    *,  # Force keywords arguments to be keyword-only
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+) -> xr.DataArray:
+    """
+    Calculates the Spearman's rank correlation coefficient between two xarray DataArrays \
+    Spearman's correlation is identical to Pearson's correlation when the relationship \
+    is linear. They diverge when the relationship is not linear as Spearman's correlation \
+    assesses monotic relationships whereas Pearson's correlation assesses strictly linear \
+    functions.
+
+
+    .. math::
+        \\rho = 1 - \\frac{6\\sum_{i=1}^{n}{d_i^2}}{n(n^2-1)}
+
+    where:
+        - :math:`\\rho` = Spearman's rank correlation coefficient
+        - :math:`d_i` = the difference between the ranks of x and y in a sample
+        - :math:`n` = the number of samples
+
+    Args:
+        fcst: Forecast or predicted variables
+        obs: Observed variables.
+        reduce_dims: Optionally specify which dimensions to reduce when
+            calculating the Spearman's rank correlation coefficient.
+            All other dimensions will be preserved.
+        preserve_dims: Optionally specify which dimensions to preserve when
+            calculating the Spearman's rank correlation coefficient. All other dimensions will
+            be reduced. As a special case, 'all' will allow all dimensions to be
+            preserved. In this case, the result will be in the same shape/dimensionality
+            as the forecast, and the errors will be the absolute error at each
+            point (i.e. single-value comparison against observed), and the
+            forecast and observed dimensions must match precisely.
+    Returns:
+        xr.DataArray: An xarray object with Spearman's rank correlation coefficient values
+
+    Note:
+        This function isn't set up to take weights.
+
+    See also:
+    `scores.continuous.correlation.pearsonr`
+
+    Reference:
+        https://en.wikipedia.org/wiki/Spearman%27s_rank_correlation_coefficient
```

Comment:
@lbluett @nicholasloveday  is this the original paper: https://doi.org/10.2307/1412159 ?

Reply:
Yes, I think that is the original paper.
I think that it may also be nice to still keep the wiki link there too as it is easier to read and provides more context.